### ISSUE TITLE
[BUG] Fix `LearningShapeletClassifier` non-standard methods

### ISSUE THREAD

**Issue by MatthewMiddlehurst** [aeon-toolkit/aeon#1387]
### Describe the bug

The non-standard methods in `LearningShapeletClassifier` were breaking the CI, likely due to the naming of `transform`. These have been commented out for now.

They should be included again and some testing should be done to figure out if this is why it breaks.

### Steps/Code to reproduce the bug

N/A

### Expected results

These methods can be included without CI breakage.

### Actual results

CI breaks.
i.e. https://github.com/aeon-toolkit/aeon/actions/runs/8560508379/job/23459706210

**Comment by nileenagp** [comment]
Hello, Just to understand the issue here, I tried to run the test_all_classifiers.py using pytest  after uncommenting "locate" and "transform" methods in LearningShapeletClassifier class. The test completed with 5 warnings and I do not see any errors raised for these methods. Can you please clarify if I am missing anything here?
![image](https://github.com/aeon-toolkit/aeon/assets/48993630/bb6bd6e6-1f47-4a8f-a8ba-dfcbb670a733)

![image](https://github.com/aeon-toolkit/aeon/assets/48993630/47e411d7-ce89-4d51-a884-2b3e347211ca)

**Comment by MatthewMiddlehurst** [comment]
Not sure. We were having failures such as https://github.com/aeon-toolkit/aeon/actions/runs/8560508379/job/23459706210 before. I think some tests check for the presence of methods such as transform.

Dont really like having these methods in a classifier either way.

**Comment by TonyBagnall** [comment]
just looking at this issue. imo these two methods (transform and locate) should only be callable after fit, and should return stored data. I'll draft a PR

**Comment by nileenagp** [comment]
I am still looking at the code to understand the test data, scenarios and the flow. When I have more clarity, hopefully I will be able to come back on this issue.

**Comment by TonyBagnall** [comment]
@nileenagp I would not put too much effort into understanding the test scenarios, we are reworking it to make it simpler. I think we can resolve this issue quite simply as described above, I'll have a go and see what you think

**Comment by nileenagp** [comment]
> @nileenagp I would not put too much effort into understanding the test scenarios, we are reworking it to make it simpler. I think we can resolve this issue quite simply as described above, I'll have a go and see what you think

Thank you for sharing the details. I checked the changes in the PR and understand what you explained here.  By looking at the code and the LearningShapelets library, as per my understanding we should still return the shapelet transform vector in the get_transform method.  Let me know if I got that wrong..

![image](https://github.com/aeon-toolkit/aeon/assets/48993630/dc215bc7-3350-4b2f-86bd-9164a0181794)

I ran the new test_ls.py that you have created with above change and logging,  it looks to be returning the shapelet transform vector as expected.

![image](https://github.com/aeon-toolkit/aeon/assets/48993630/eda0d3f7-d4b7-45ba-9eda-59d0391cb2f0)
